homeassistant/components/thermostat/honeywellus.py

```diff
@@ -0,0 +1,169 @@
+"""
+Honeywell US Thermostat Support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Adds support for Honeywell connected thermostats sold in the US
+
+Configuration:
+
+thermostat:
+  platform: honeywellus
+  username: USERNAME
+  password: PASSWORD
+  id: IDENTIFIER
+
+Variable:
+
+username
+*Required
+USERNAME should be your username for the honeywell totalcomfort website
+
+password
+*Required
+PASSWORD should be your password for the honeywell totalcomfort website
+
+IDENTIFIER
+*Required
+IDENTIFIER is the numeric identifier for the thermostat you want to control
+"""
+import logging
+import requests
+
+from homeassistant.components.thermostat import ThermostatDevice
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, TEMP_CELCIUS,
+                                 TEMP_FAHRENHEIT)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SYSTEM_SWITCH_POSITIONS = {1: 'Heat',
+                           2: 'Off',
+                           3: 'Cool'}
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Configure Honeywell US thermostats. """
+    ident = config.get('id')
+    usern = config.get(CONF_USERNAME)
+    passw = config.get(CONF_PASSWORD)
+
+    if ident is None or usern is None or passw is None:
+        _LOGGER.error('A thermostat id, username and password are required')
+        return False
+
+    add_devices([HoneywellUSThermostat(ident, usern, passw)])
+
+
+class HoneywellUSThermostat(ThermostatDevice):
+    """ Represents a Honeywell US Thermostat. """
+
+    def __init__(self, ident, username, password):
+        self._ident = ident
+        self._username = username
+        self._password = password
+        self._session = requests.Session()
+        # Yeah, really.
+        self._session.headers['X-Requested-With'] = 'XMLHttpRequest'
+        self._update()
+
+    def _login(self):
+        self._session.get('https://mytotalconnectcomfort.com/portal/')
```

Comment:
You want to add timeout to all your requests so that it won't lock up a thread in Home Assistant when things go wrong.